apple/swift-docc-render
# Don't render default footer for IDE target

Bug/issue #, if applicable: 89998805

## Summary

Fixes an issue where the default footer is incorrectly rendered for IDE targets in certain edge case scenarios.

Right now, this is happening when navigating very quickly between pages where occasionally there is an error encountered when attempting to load data, which is handled by calling `next(false)` in a vue router guard to abort the navigation—when this happens, a blank page is loaded, but the footer is still shown unexpectedly.

_Note: marking this as a draft PR for the moment, since I am still doing some of my own manual testing to verify that these changes resolve the issue without introducing any new regressions._

## Testing

Steps:
1. Run `VUE_APP_TARGET=ide npm run serve`
2. Open http://localhost:8080/documentation/foobar and verify that a blank page is shown with no color scheme toggle footer 
3. Run `VUE_APP_DEV_SERVER_PROXY=https://ethan-kusters.github.io/swift-argument-parser npm run serve`
4. Open http://localhost:8080/documentation/argumentparser and verify that the color scheme toggle footer is still shown
5. Open http://localhost:8080/documentation/foobar and verify that the color scheme toggle is still shown

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.
